### ISSUE TITLE
Send email notification about share for group shares

### ISSUE THREAD

**Issue by juliushaertl** [nextcloud/server#13158]
Similar to user shares from https://github.com/nextcloud/server/pull/5897 we should also send out notification emails when sharing a file or folder to a group.

**Comment by juliushaertl** [comment]
@nextcloud/sharing

**Comment by schiessle** [comment]
not sure about it. This is a hard coded behavior at the moment for user shares which can't be changed in any settings, afaik. Also it makes the activity settings where the user can decide by their own if they want to get a mail notification or not useless.

**Comment by juliushaertl** [comment]
Maybe we could just send activity mails for file shares directly so we don't have a delay for them? We wouldn't need the hardcoded notification mail then at all, right?

**Comment by rullzer** [comment]
The problem is that this doesn't scale. We kind of need this to run in a cron job. Else on big instances this will timeout. sharing to a group with 1k users etc.

**Comment by schiessle** [comment]
@rullzer yes, scaling is another issue. I just looked it up and it seems that you can have max 50 recipients per mail so for large groups this is a problem as well. But beside that I don't like the idea to work around the activity settings, because I think it makes way more sense to let the user define what kind of mail they want to get.

**Comment by eatyourpeas747** [comment]
Any progress on this feature?
